### DISCUSSION TITLE
基于内置qiankun实现微前端，其中主子应用都是umi4，但引入子应用导致主应用主题被覆盖，请问有什么办法可以解决？

### DISCUSSION THREAD

**Discussion by tpc-ht** [umijs/umi#11412] [Q&A] [answered]
使用
https://umijs.org/docs/max/introduce#%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8
安装

**Comment by fz6m** [comment, marked as answer]
要看你的主题方案是啥，如果是 antd 5 的话，css in js 是与 design token 有关的，每个区域可以用独立的 `ConfigProvider` 自定义主题，这是最推荐的。

如果是 css 变量这种，因为 qiankun 默认没有开样式隔离，所以子应用加载出来，子应用的样式就外露了，如果把 qiankun 样式隔离打开并不好（如弹窗不到中间、性能缓慢等，不推荐），考虑给不同的应用加 class 前缀吧，比如 antd 4 可以配 class prefix ，每个应用都独立就不会冲突了，自己研究下。

**Reply by tpc-ht** [reply]
![image](https://github.com/umijs/umi/assets/59469126/4e6d0043-5b85-4085-aa71-3d1bb67bd5e9)
用的是antd4，主组件在config.ts中配置了全局的样式变量。子应用因为是新的，所以 theme 为空，在加载子应用的时候，导致主应用的整体样式发生改变。
目前将子应用的主题设置成一样的就不会。这是最优解决方案吗？

**Reply by fz6m** [reply]
自定义一下 `@ant-prefix` 前缀，然后导入 less ，这样产物里 antd 的样式类名是完全不同的（需要注意全局复写的地方会失效），自己找相关资料、文档、文章研究下吧。
